optimum/habana/transformers/models/llama/modeling_llama.py

```diff
@@ -364,17 +364,24 @@ def pre_attn_forward(
                 else:
                     kv_seq_len += past_key_value[0].shape[-2]
             else:
-                if reuse_cache:
+                if reuse_cache and not isinstance(past_key_value[0], torch.Tensor):
                     kv_seq_len = past_key_value[0][-2]
                 else:
-                    kv_seq_len = past_key_value[0].shape[-2]
+                    kv_seq_len = (
+                        past_key_value[0].shape[-2] + kv_seq_len
+                        if past_key_value[0].shape[-2] < token_idx.cpu()
```

Comment:
this could cause tgi performance issue as token_idx needs to be moved to cpu each time. I would suggest to have model argument